Automattic/simplenote-macos
# Refreshing Markdown Preview onChange

### Details:
In this PR we're implementing a mechanism that will allow us to dynamically refresh the Markdown Preview UI, whenever the entity onscreen gets updated.

@etoledom Eduardoooooo!! You game for a quick PR?
Thanks in advance!!!

Closes #642

### Test
1. Add a new document
2. Make some random edits. **Please** add text on the upper side, move the cursor, add more text. Repeat few times
 2.1. Changing the cursor position generates an "Undo Checkpoint" in NSUndoManager!
3. Press the `( ... )` button and enable **Markdown**
4. Press the Markdown Preview button 👁️ 

- [ ] Verify that pressing CMD + Z gets the Markdown Preview updated

### Release
`RELEASE-NOTES.txt` was updated in 9cd0c8b with:

> Markdown Preview will now refresh, whenever the Note is updated
